src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1164,14 +1174,13 @@ LogicalResult ONNXBatchNormalizationTestModeOp::inferShapes() {
   // Check whether the shapes of scale, bias, mean and variance are valid.
   // Operand's dimensions can be in the form of NxCxD1xD2x...xDn or N.
   // In case of N, C is assumed to be 1.
+  // 2-D tensors are assumed to be of shape NxC
   // Shapes of scale, bias, mean and variance must be C.
   int64_t c = -1;
   if (inputTensorTy.getShape().size() == 1) {
     c = 1;
-  } else if (inputTensorTy.getShape().size() > 2) {
+  } else if (inputTensorTy.getShape().size() >= 2) {
     c = (inputTensorTy.getShape()[1] != -1) ? inputTensorTy.getShape()[1] : -1;
-  } else {
-    return emitError("Wrong rank for the input");
```

Comment:
Why was rank == 2 a problem before? @AlexandreEichenberger @tjingrant @tungld

Reply:
We never have an input of rank 2 as described in ONNX (https://github.com/onnx/onnx/blob/master/docs/Operators.md#BatchNormalization). Only N or NxCxD1xD2x...xDn.

Reply:
We do see networks that have this though...exported from pytorch (which is why we needed to change this)

Reply:
Have you confirmed that the onnx model exported from pytorch is runnable, say, with onnxruntime? It is helpful if you can share the model here. I just wanted to be convinced about this case since the ONNX document does not mention it. Thanks!

Reply:
That would be great! Thanks a lot for looking into that!

Reply:
From looking in the code I don't see anything obvious that would break. I'm trying to add a test now, but I'm getting git submodule issues so I will have to resolve those before I can add the test.

Reply:
@doru1004 I added a 2D batchnorm test and made the changes in Conversion/NN/Normalization to make it work (it was a very simple modification) - let me know what you think?